test/e2e/syncer/syncer_test.go

```diff
@@ -170,6 +170,30 @@ func TestSyncerLifecycle(t *testing.T) {
 	}
 
 	// TODO(marun) Check that the deployment was able to contact kcp
+
+	//Delete the deployment
+	err = downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Delete(ctx, deployment.Name, metav1.DeleteOptions{})
+	require.NoError(t, err)
+
+	//Wait the deployment to be recreated and check it is a different UID
+	t.Logf("Waiting for downstream deployment %s/%s to be created...", downstreamNamespaceName, upstreamDeployment.Name)
+	require.Eventually(t, func() bool {
+		newDeployment, err := downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false
+		}
+		if err != nil {
+			t.Errorf("saw an error waiting for downstream deployment %s/%s to be created: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
+		}
+		//Test UID
+		if deployment.UID == newDeployment.UID {
+			return false
+		}
+		//Set the deployment to the new created one
+		deployment = newDeployment
+		return true
+	}, wait.ForeverTestTimeout, time.Second*10, "downstream deployment %s/%s was not synced", downstreamNamespaceName, upstreamDeployment.Name)
```

Comment:
10s is a very long poll. 100ms please

Reply:
100ms is not enough, I will try to shorten it

Reply:
Please use 100ms. This is the polling interval, not the timeout.

Reply:
300ms seems ok

Reply:
If 100ms is not working, something is wrong inside the eventually block